src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/TraceContextInjection/Integrations/WriteToTargetsIntegration.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.TraceContextInjection.Integrations;
+
+/// <summary>
+/// NLog integration for NLog 5.x (with ITargetWithFilterChain).
+/// This integration intercepts NLog's internal WriteToTargets method to:
+/// 1. Inject trace context (TraceId, SpanId, TraceFlags) into the LogEventInfo properties
+/// 2. Forward log events to OpenTelemetry when the bridge is enabled
+/// </summary>
+/// <remarks>
+/// NLog 5.x has assembly version 5.0.0.0 regardless of the NuGet package version.
+/// Later NLog 5.x versions (5.3.0+) use the ITargetWithFilterChain interface.
+/// Both this integration and WriteToTargetsLegacyIntegration are registered for
+/// NLog 5.x to handle both the interface and concrete class variants.
+/// The native profiler will match the correct integration based on the actual
+/// method signature at runtime.
+/// </remarks>
+[InstrumentMethod(
+    assemblyName: "NLog",
+    typeName: "NLog.Logger",
+    methodName: "WriteToTargets",
+    returnTypeName: ClrNames.Void,
+    parameterTypeNames: ["NLog.LogEventInfo", "NLog.Internal.ITargetWithFilterChain"],
+    minimumVersion: "5.0.0",
+    maximumVersion: "5.*.*",
+    integrationName: "NLog",
+    type: InstrumentationType.Log)]
+public static class WriteToTargetsIntegration
```

Comment:
Similar to what I have posted for 6.0.0
Two independent methods needs to be covered
* `private void WriteToTargets(Type wrapperType, [NotNull] LogEventInfo logEvent, [NotNull] ITargetWithFilterChain targetsForLevel)` - not covered
* `private void WriteToTargets([NotNull] LogEventInfo logEvent, [NotNull] ITargetWithFilterChain targetsForLevel)` - covered